src/mavsdk/plugins/telemetry/telemetry_impl.cpp

```diff
@@ -1176,6 +1181,42 @@ void TelemetryImpl::process_battery_status(const mavlink_message_t& message)
     }
 }
 
+void TelemetryImpl::process_battery_status_v2(const mavlink_message_t& message)
+{
+    mavlink_battery_status_v2_t bat_status;
+    mavlink_msg_battery_status_v2_decode(&message, &bat_status);
+
+    _has_bat_status = true;
+
+    Telemetry::Battery new_battery;
+    new_battery.id = bat_status.id;
+    new_battery.voltage_v = bat_status.voltage; 
+    //Is it correct to set this to NaN for UINT32_MAX? HOw do you specify = NaN?
+
+    // FIXME: it is strange calling it percent when the range goes from 0 to 1.
+    // Can we fix this? What about if value not provided?
```

Comment:
See this fixme? Would be good to break this and send a percentage at some point. How are you managing those kinds of changes?

Reply:
Yes, there are a few that are wrong like that. Thanks for pointing it out. Version 2 is coming soon, so now is the time to break/fix it.

Here is where it needs fixing first:
https://github.com/mavlink/MAVSDK-Proto/blob/e8526d1fd8505fa3bc47eb1ea5edb31e58cb3be0/protos/telemetry/telemetry.proto#L571

Reply:
@julianoes Thanks, but I think I need a bit more info.
This is what it looks like now:
```
    float remaining_percent = 2 [(mavsdk.options.default_value)="NaN"]; // Estimated battery remaining (range: 0.0 to 1.0)
```
1. As I understand it, to allow us to break this, first I change the name. So 
   ```
   float percent_remaining = 2 [(mavsdk.options.default_value)="NaN"]; // Estimated battery percentage remaining (range: 0 to 100).
   ```
   - What is the value after `=`? In this case the value is 2. My guess would be the default initialisation, but then we afterwards state `mavsdk.options.default_value`=NaN. That would imply to me that the default is NaN?
   - This is sent as a uint8. It's an integer with an invalid value of UINT8_MAX. Can we do something like (?):
     ```
     uint8 percent_remaining = 255 [(mavsdk.options.default_value)="UINT8_MAX"]; // Estimated battery percentage remaining (range: 0 to 100).
     ```

Reply:
Not exactly :slightly_smiling_face:. I believe that the only thing to change there is the docs (`((range: 0.0 to 1.0)` to `(range: 0 to 100)`). Let me explain:

* `remaining_percent` follows the convention that the "unit" goes at the end, like `velocity_m_s` and `time_s`. I don't have a strong opinion here so if you insist, we can rename it :+1:.
* The number after `=` is for protobuf. In protobuf, each field has an "id", which is used in the binary representation. Something like the `msgid` in MAVLink, I guess. We never want to change it.
* `[(mavsdk.options.default_value)="NaN"]` This is not used by protobuf, just by our MAVSDK code generators. The thing is that "UINT8_MAX" is a C definition, and this value could be used by the templates in all languages. We could have the templates translate that to the right value for each language, but... why would we set a float default value to uint8_max? Feels like `NaN` makes more sense for an undefined float :thinking:.

Reply:
I would say `_percent` should go from 0..100. If there is no unit, it should go from 1.0. I know this might not be inline with what I have implemented earlier (case and point here) but I'd say that's the most intuitive.

Reply:
I agree with @JonasVautherin that we can't have magic numbers in the API except NAN.

Reply:
@julianoes @JonasVautherin So the reason I wanted to change the name was that Julian said that we change the name on API break - I presume the assumption being that it fails a compile and users check what's changed.

So I am happy to change `(range: 0.0 to 1.0) ` to `(range: 0 to 100)` but don't we need to rename it something? I could call it `charge_percent` if you prefer?

Reply:
> that we change the name on API break

We _can_ change the name on API break if we want to. It's just an opportunity, it's not mandatory. I think there was a misunderstanding there, right? :grin:

It's just that for semantic versioning, every time we break the API, we have to increase the major version. So we don't want to be on MAVSDK v503 just because we rename stuff :grin:. However, we will soon move to MAVSDK v2 (we had a real need to break the API), so that's an opportunity to rename small stuff (and break more of the API on this occasion :laughing:).

Does that make sense?